### PR TITLE
[MM-25263] Add selector to get profiles in group

### PR DESCRIPTION
**Pull request by fmunshi**
#### Summary
- Modify the groups redux store to have members be loaded into the users so that we can re-use existing search logic.
- Adds redux for group stats since we will no longer be calling the group members endpoint

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-25263

### CODE REVIEW

File: src/action_types/groups.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export default keyMirror({
     RECEIVED_GROUP_TEAMS: null,
     RECEIVED_GROUP_CHANNELS: null,
 
-    RECEIVED_GROUP_MEMBERS: null,
```
Comment (mkraft) [inline src/action_types/groups.ts:15]: I take it nobody is listening for `RECEIVED_GROUP_MEMBERS` this anymore? Neither web nor mobile?

File: src/selectors/entities/users.ts [hunk 2]
```diff
@@ -562,3 +567,22 @@ export function makeGetDisplayName(): (state: GlobalState, userId: $ID<UserProfi
...
+    },
+);
+
+export function searchProfilesInGroup(state: GlobalState, groupId: $ID<Group>, term: string, skipCurrent = false, filters?: Filters): Array<UserProfile> {
+    const profiles = filterProfilesMatchingTerm(getProfilesInGroup(state, groupId, filters), term);
+    if (skipCurrent) {
+        removeCurrentUserFromList(profiles, getCurrentUserId(state));
+    }
+
+    return profiles;
```
[the remark this answers is not in the document]
Comment (mkraft) [reply src/selectors/entities/users.ts:587]: @hahmadia Those aren't equivalent and would only work if `removeCurrentUserFromList` were updated to return a list vs mutating the input list.

File: src/action_types/groups.ts (hunk 1 above)
Comment (fmunshi) [reply src/action_types/groups.ts:15]: yup

File: src/selectors/entities/users.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (hahmadia) [reply src/selectors/entities/users.ts:587]: Then nevermind what I said

**Comment by srkgupta** [comment]
Hi @fmunshi 
I am ready to pick this PR for testing. Can you please resolve the merge conflicts. Thanks.

**Comment by fmunshi** [comment]
@srkgupta I have resolved the merge conflicts here and updated the webapp with the latest redux hash

**Pull request merged**